### PR TITLE
Adding an HTTP Listener

### PR DESCRIPTION
**Pull request by dbemiller**
This follows up on #404 by adding a file which populates the Stored Request caches via HTTP endpoints.

All the Stored Requests are fetched on startup, and then refreshed periodically.

### CODE REVIEW

File: stored_requests/events/events.go [hunk 1]
```diff
@@ -60,10 +60,10 @@ func (e *EventListener) Stop() {
 func (e *EventListener) Listen(cache stored_requests.Cache, events EventProducer) {
 	for {
 		select {
-		case update := <-events.Updates():
-			cache.Update(context.Background(), update.Requests, update.Imps)
-			if e.onUpdate != nil {
-				e.onUpdate()
+		case save := <-events.Saves():
+			cache.Save(context.Background(), save.Requests, save.Imps)
+			if e.onSave != nil {
```
Comment (cirla) [inline stored_requests/events/events.go]: Should we remove the Update method from the Cache altogether if it's not being used or should we expose Save events in addition to Update events and have producers decide which to use?

File: stored_requests/events/http/http.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	return e
+}
+
+type httpEvents struct {
+	client        *httpCore.Client
+	ctxProducer   func() (ctx context.Context, canceller func())
+	endpoint      string
+	invalidations chan events.Invalidation
+	lastUpdate    time.Time
+	updates       chan events.Save
```
Comment (cirla) [inline stored_requests/events/http/http.go]: If we're removing updates in favor of saves there are a few instances in this file where `updates` is referenced where it's actually a save.

File: stored_requests/events/events.go (hunk 1 above)
Comment (dbemiller) [reply stored_requests/events/events.go]: yeah, I'm fine with that. Dead code is almost always more confusing than useful. "You ain't gonna need it" and all that.

File: stored_requests/events/http/http.go (hunk 2 above)
Comment (dbemiller) [reply stored_requests/events/http/http.go]: good call, thanks... updated now.

File: stored_requests/events/events.go (hunk 1 above)
Comment (dbemiller) [reply stored_requests/events/events.go]: I chatted with Hans offline... we agreed that deleting was fine.

**Pull request merged**